### PR TITLE
CI: run `git fetch` before checking for docs-only change

### PR DESCRIPTION
**Pull request by chrismwendt**
Fixes https://github.com/sourcegraph/sourcegraph/issues/2823

### CODE REVIEW

File: enterprise/dev/ci/gen-pipeline.go [hunk 1]
```diff
@@ -60,6 +60,11 @@ func main() {
 		bk.Env("DATE", now.Format(time.RFC3339)),
 	)
 
+	_, err := exec.Command("git", "fetch").Output()
```
Comment (slimsag) [inline enterprise/dev/ci/gen-pipeline.go]: It's very odd to use `exec.Command` here. That means running `gen-pipeline.go` which should just generate and output the buildkite yaml is suddenly doing CI work aside from generating yaml

Comment (chrismwendt) [reply enterprise/dev/ci/gen-pipeline.go]: Agreed, switched to a `.buildkite/hooks/post-checkout` [Buildkite repository hook](https://buildkite.com/docs/agent/v3/hooks#repository-hooks).

File: .buildkite/hooks/post-checkout [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+echo "Running git fetch..."
```
Comment (slimsag) [inline .buildkite/hooks/post-checkout]: You should add a comment explaining why we do this. At first glance, it won't make sense to anyone else without an explanation: "buildkite is literally for running CI and this is a post checkout hook.. why wouldn't buildkite checkout the right revision?"

**Comment by slimsag** [comment]
LGTM but explanation comment is important I think

File: .buildkite/hooks/post-checkout (hunk 2 above)
Comment (chrismwendt) [reply .buildkite/hooks/post-checkout]: :+1: Adding a comment in https://github.com/sourcegraph/sourcegraph/pull/2827

**Pull request merged**